src/Workspaces/Remote/Core/Services/RemoteWorkspace.cs

```diff
@@ -57,7 +57,6 @@ public Solution AddSolution(SolutionInfo solutionInfo)
             lock (_gate)
             {
                 this.OnSolutionAdded(solutionInfo);
-                this.UpdateReferencesAfterAdd();
```

Comment:
Can we add a stronger guard to prevent this method being invoked in the RemoteWorkspace in future? Probably an internal virtual property on Workspace, which is true by default, and false for this workspace, and add a contract check in UpdateReferencesAfterAdd for this property?